Updating Code based on Additional Linters

kensipe opened the pull request:
Slowing adding a couple of linters listed beflow, there are a number of lint issues to resolve.    Fixed in this PR:
1. some `goconst` around existing consts that were not used
2. `unconvert` for unnecessary conversations

history:   The current project configuration is such that a user with a `.golangci-lint.yml` in their home dir would override existing configurations.   Remaining on this PR are reasonable noncontroversial updates to code based on some additional linters (unknownly at the time) being applied.   There is another PR coming to change to a configuration which will eliminate this environmental issue.

Signed-off-by: Ken Sipe <kensipe@gmail.com>

BenTheElder commented:
hmm, the linters should be running and passing in CI already?

BenTheElder commented on pkg/build/nodeimage/build_impl.go:
```diff
@@ -349,6 +329,32 @@ func (c *buildContext) prePullImages(bits kube.Bits, dir, containerID string) ([
 	return importer.ListImported()
 }
 
+func pullImageFns(requiredImages []string, builtImages sets.String, imagesDir string, pulledImages chan string, logger log.Logger) []func() error {
```
I'm -1 on this, this loses a ton of context, this method takes way too many parameters.

this whole file does need refactoring, but I'd rather not just do enough to silence some linter.
we're going to continue restructuring build to solve some open issues as-is.

kensipe replied:
yeah... I thought this might be controversial

I'll revert and mark ignore for this method

BenTheElder commented on pkg/cluster/internal/kubeconfig/internal/kubeconfig/paths.go:
```diff
@@ -40,7 +40,7 @@ https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands
 
 - Otherwise, ${HOME}/.kube/config is used and no merging takes place.
-*/
+*/ // nolint:lll
```
this linter should probably just be disabled. we're not using 80 character-width hardware terminals in 2020.

kensipe replied:
totally align with you there!  on it

BenTheElder commented on pkg/cluster/internal/providers/docker/network.go:
```diff
@@ -18,7 +18,7 @@ package docker
 import (
 	"bytes"
-	"crypto/sha1"
+	"crypto/sha1" //nolint:gosec // used as a seed to subnet generator
```
I think we should probably just disable this linter wholesale, I'm not aware of it catching much useful and it complains about a lot of intentional, well-meant behavior

kensipe replied:
roger that

BenTheElder reviewed (commented):
we should verify that the verify job is doing it's job, lint passing is supposed to be enforced in presubmit.

I'm not a fan of linter-comments, but if we're going to leave linter comments we should leave an explanation of why, so in the future it can be removed if appropriate.
In all cases here I think we should disable the linter, they don't seem particularly useful.

kensipe commented:
@BenTheElder thanks for the guidance and context.  I updated the title and desc of this PR with current details.  Also squashed during rebase and ready.
I will be pulling together a PR specific to linter configuration in a yaml config file NOT in the root of the project.

kensipe commented:
/retest
